### ISSUE TITLE
When making a call from mobile network the media setup fails with ICE Connection Failed

### ISSUE THREAD

**Issue by atsakiridis** [RestComm/restcomm-ios-sdk#238]
I think the problem is the candidates:

a=candidate:66916376 1 udp 2122260223 100.105.101.20 58004 typ host generation 0
a=candidate:66916376 2 udp 2122260222 100.105.101.20 63616 typ host generation 0
a=candidate:3480235915 1 udp 1686052607 62.74.1.17 39412 typ srflx raddr 100.105.101.20 rport 58004 generation 0
a=candidate:3480235915 2 udp 1686052606 62.74.1.17 39413 typ srflx raddr 100.105.101.20 rport 63616 generation 0
a=candidate:1300163816 1 tcp 1518280447 100.105.101.20 58449 typ host tcptype passive generation 0
a=candidate:1300163816 2 tcp 1518280446 100.105.101.20 58450 typ host tcptype passive generation 0 

I'd expect to have 62.* candidates for TCP as well in order for the scenario to work

**Comment by atsakiridis** [comment]
Important note: the issue only occurs when calling Olympus behind NAT. When calling Restcomm/MS everything is ok

**Comment by atsakiridis** [comment]
Did some more digging and I'm starting to arrive at the conclusion that we won't be able to have proper media flowing over Cellular network unless we implement TURN. I did two tests with google's apprtcdemo iOS app (to rule out an bug in our SDK):

- Call between browser and iOS using wifi
- Call between browser and iOS using 3g

From the capture, it seems that when using wifi, media is transferred locally. When using 3g it is using TURN. In 3g I also checked that the STUN pings towards 'host' and 'reflex' candidates are timing out (which is what I saw in our SDK as well), and later a TURN data channel is created and used for the data transfer.

Also found these: 

- https://code.google.com/p/webrtc/issues/detail?id=2239
- http://stackoverflow.com/questions/21155888/a-stun-and-turn-server-that-work-under-3g-4g-network

After discussions with @deruelle we decided to add a configuration option to enable TURN (default off) and at some point implement trickle ICE to avoid to huge delay when using TURN but the other peer doesn't.
